api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -57,10 +58,14 @@ public String getIdentifierForSeed(long seed) {
 
 		String identifier = IdgenUtil.convertToBase(seed, baseCharacterSet.toCharArray(), seqLength);
 
-    	// Add optional prefix and suffix
-    	identifier = (prefix == null ? identifier : prefix + identifier);
+    	if (prefixProviderBean != null) {
+    		prefix = getPrefixProvider(prefixProviderBean).getValue();
+    		identifier = (prefix == null ? identifier : prefix + identifier);
+    	} else {
+    		identifier = (prefix == null ? identifier : prefix + identifier);
+    	}
     	identifier = (suffix == null ? identifier : identifier + suffix);
```

Comment:
See above:  we could have this logic interpret the "prefix" column and operate accordingly.